content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +323,78 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+When a new NetworkPolicy object is created, it may take some for a network plugin
+to handle the new object. If the pod that is affected by a NetworkPolicy
+is created before the network plugin has completed NetworkPolicy handling,
+that pod may be started unprotected, and isolation rules will be applied when 
+the NetworkPolicy handling is completed.
+
+Once the NetworkPolicy is handled by a network plugin,
+
+1. All newly created pods affected by a given NetworkPolicy will be isolated before 
+they are started.
+
+2. Allow rules will be applied eventually after the isolation rules (that includes applying both at the same time).
+In the worst case, a newly created pod may have no network connectivity at all, if
+isolation rules were already applied, but no allow rules were applied yet.
+
+Every created NetworkPolicy will be handled by a network plugin eventually, but there is no
+way to tell from the Kubernetes API when exactly that happens. 
+The only way to check if the network policy is applied to a pod, is to run a connectivity check
+from that pod.
```

Comment:
Even then, you might not be able to tell (I think).

Reply:
that is right, e.g. connections may be dropped for another reason, but I am not sure if adding these details will improve or over-complicate this section, wdyt?

Reply:
I'd omit the sentence. I'm wary about the risk of misleading people.

(I don't want someone to think “_Right, so this Pod shouldn't be able to talk to a Pod in namespace `untrusted`. I tried and failed, so my policy is working. Job done, :shipit:!_”)